Make dot and ls dependencies work without ghc installed (#4405)
commercialhaskell/stack: ChangeLog.md

  [#4314](https://github.com/commercialhaskell/stack/pull/4314)
* Add `--cabal-files` flag to `stack ide targets` command.
* Don't download ghc when using `stack clean`.

> * `dot` and `ls dependencies` commands is no longer require ghc installed. And, fix not respected `--no-install-ghc` flag. (See [#4390](https://github.com/commercialhaskell/stack/issues/4390))

Reviewer: ```suggestion
* `dot` and `ls dependencies` commands no longer require GHC to be installed. Also, ensures the `--no-install-ghc` flag is respected. See: [#4390](https://github.com/commercialhaskell/stack/issues/4390)
```

> * `dot` and `ls dependencies` commands no longer require GHC to be installed. Also, ensures the `--no-install-ghc` flag is respected. See: [#4390](https://github.com/commercialhaskell/stack/issues/4390)

Reviewer 2: I think that 2nd second sentence is misleading, the flag was respected at least in version 1.7.1 already:
```
qrilka@qdesktop ~/ws/h/stack-test-ghc-7.8.4 $ cat stack.yaml 
resolver: ghc-7.8.4
qrilka@qdesktop ~/ws/h/stack-test-ghc-7.8.4 $ stack --no-install-ghc clean
No compiler found, expected minor version match with ghc-7.8.4 (x86_64-tinfo6) (based on resolver setting in /home/qrilka/ws/h/stack-test-ghc-7.8.4/stack.yaml).
To install the correct GHC into /home/qrilka/.stack/programs/x86_64-linux/, try running "stack setup" or use the "--install-ghc" flag.
qrilka@qdesktop ~/ws/h/stack-test-ghc-7.8.4 $ stack --numeric-version
1.7.1
```

> * `dot` and `ls dependencies` commands no longer require GHC to be installed. Also, ensures the `--no-install-ghc` flag is respected. See: [#4390](https://github.com/commercialhaskell/stack/issues/4390)

Author: It's perfectly clear, thanks!!
